app/static/js/userProfile.js

```diff
@@ -61,6 +61,45 @@ $(document).ready(function(){
 
   });
 
+  $("#addNoteButton").click(function() {
+    $("#noteModal").modal("toggle");
+  });
+
+  $("#bonnerInput").click(function() {
+    $("#bonnerInput").val(
+      $("#bonnerInput").val()== "on" ? "off" : "on"
+    )
+  });
+
+  $('#addNoteForm').submit(function(event) {
+    event.preventDefault()
+    let username = $("#notesSaveButton").data('username')
+    $.ajax({
+      method: "POST",
+      url:  "/profile/addNote",
+      data: {"username": username,
+             "visibility": $("#noteDropdown").val(),
+             "noteTextbox": $("#addNoteTextArea").val(),
+             "bonner": $("#bonnerInput").val()=="on"? "on":"off"},
```

Comment:
Is there a reason to keep using 'on' and 'off' instead of just passing true or false to the controller?